### PR TITLE
#337 save student detailes will not back to student list

### LINKED ISSUE #337: בסיום שמירת פרטים אישיים יש להשאר באותו מסך עם חיווי מתאים ("Vנשמר" על הכפתור וחיווי תוך כדי שמירה (במידת הצורך) - כפי שמוגדר ב-UI) ולא לחזור לרשימת התלמידים, זאת על מנת לאפשר למשתמש להמשיך למסך מערכת שעות רק במעבר טאב
child of #329

### LINKED ISSUE #339: עידכונים בפרטים אישיים לא מתרפרשים במסך ניהול תלמידים (מופיעים הפרטים הקודמים)
child of #329

### PR DESCRIPTION
**Pull request by ulmanA**
fixes #337 
fixes #339

### CODE REVIEW

**Comment by pelegbahar** [comment]
@ulmanA can you please resolve the coverage issue?

File: src/app/pages/student/details/tabs/student-details-personal-info/student-details-personal-info.component.ts [hunk 1]
```diff
@@ -16,14 +16,15 @@ export class StudentDetailsPersonalInfoComponent implements OnInit {
   classes: Class[];
   isNewStudent: boolean;
   idOrNew: string;
+  changesWhereSaved = false;
```
Comment (tulitheprogrammer) [inline src/app/pages/student/details/tabs/student-details-personal-info/student-details-personal-info.component.ts]: Hi @ulmanA , please fix if this is a typo (changesWereSaved)

File: src/app/pages/student/details/tabs/student-details-personal-info/student-details-personal-info.component.ts [hunk 2]
```diff
@@ -81,24 +82,31 @@ export class StudentDetailsPersonalInfoComponent implements OnInit {
 
   async addStudent() {
     try {
+      this.saveFailed = false;
       await this.studentService.create(this.student);
-      this.router.navigate(['/student']);
+      this.changesWhereSavedNotification();
```
Comment (tulitheprogrammer) [inline src/app/pages/student/details/tabs/student-details-personal-info/student-details-personal-info.component.ts]: Hi @ulmanA , please fix if this is a typo (changesWereSavedNotification)

File: src/app/pages/student/details/tabs/student-details-personal-info/student-details-personal-info.component.ts [hunk 3]
```diff
@@ -81,24 +82,31 @@ export class StudentDetailsPersonalInfoComponent implements OnInit {
...
     } catch (error) {
-      // TODO: implement error handling on UI
-      console.error('Error handling not implemented');
-      throw error;
+      this.saveFailed = true;
+      console.error('Error on update student', error);
     }
   }
+
+  changesWereSavedNotification() {
```
Comment (eran1205) [inline src/app/pages/student/details/tabs/student-details-personal-info/student-details-personal-info.component.ts]: I don't like solutions that based on timeout. Can you explain what this is doing? Can we have other code than timeout?

Comment (ulmanA) [reply src/app/pages/student/details/tabs/student-details-personal-info/student-details-personal-info.component.ts]: @eran1205 
It changes the button to נשמר with green background only for 1000ms.
After the timout, the button will return to its original design.

**Pull request merged**